net/socket/accept.c

```diff
@@ -259,29 +263,31 @@ int accept(int sockfd, FAR struct sockaddr *addr, FAR socklen_t *addrlen)
       goto errout;
     }
 
+  ret = psock_accept(psock, addr, addrlen, &newsock);
```

Comment:
this doesn't look safe as accept can record the reference to newsock somewhere.
eg. tcp_start_monitor

Reply:
Good, catch.